src/dune/opam_create.ml

```diff
@@ -114,6 +114,17 @@ let insert_dune_dep depends dune_version =
       dune_dep :: List.rev acc
     | (dep : Package.Dependency.t) :: rest ->
       if Package.Name.equal dep.name dune_name then
+      let get_version s =
+          match String.split_on_char ~sep:'.' s with
+          | major :: minor :: _-> (int_of_string major, int_of_string minor)
+          | [major] -> (int_of_string major, 0)
+          | [] -> dune_version in
+        let () = match dep.constraint_ with
+        | Some (Uop (_, QVar version)) when get_version version < dune_version -> User_warning.emit ~hints:[Pp.textf "Set dune constraint to >= %s" (Dune_lang.Syntax.Version.to_string dune_version) ] [
```

Comment:
Could you run `$ make fmt` on this commit? It's a little hard to read otherwise.

The code looks OK as far as I can tell though.